server/src/main/java/org/candlepin/model/EntitlementCurator.java

```diff
@@ -363,16 +363,59 @@ private boolean poolOverlapsRange(Pool p, Date startDate, Date endDate) {
         return page;
     }
 
+    /**
+     * Deletes the given entitlement.
+     *
+     * @param entity
+     *  The entitlement entity to delete
+     */
     @Transactional
     public void delete(Entitlement entity) {
         Entitlement toDelete = find(entity.getId());
-        log.debug("Deleting entitlement: {}", toDelete);
-        log.debug("certs.size = {}", toDelete.getCertificates().size());
 
-        for (EntitlementCertificate cert : toDelete.getCertificates()) {
-            currentSession().delete(cert);
+        if (toDelete != null) {
```

Comment:
The entitlement entity given to the method can still be not-null, but the resolution fails. If there's nothing to delete, then we fail out with null pointer exceptions instead of doing nothing.